src/main/java/life/qbic/data/processing/scanner/Scanner.java

```diff
@@ -0,0 +1,147 @@
+ * currently uploaded must not be processed, which will lead to corrupt data.
+ *
+ * @since 1.0.0
+ */
+public class Scanner extends Thread {
+
+  private static final Logger log = getLogger(Scanner.class);
+  private static final Path REGISTRATION_PATH = Paths.get("registration");
+
+  private final Path scannerPath;
+  private final int scanInterval;
+  private final HashSet<Path> userProcessDirectories = new HashSet<>();
+  private final ConcurrentRegistrationQueue registrationQueue;
+  private final HashSet<RegistrationRequest> submittedRequests = new HashSet<>();
+  private final Path usersErrorDirectory;
+
+  public Scanner(ScannerConfiguration scannerConfiguration,
+      ConcurrentRegistrationQueue registrationQueue, GlobalConfig globalConfig) {
+    this.setName("Scanner-Thread");
+    Objects.requireNonNull(scannerConfiguration, "scannerConfiguration must not be null");
+    scannerPath = Path.of(scannerConfiguration.scannerDirectory());
+    if (!scannerPath.toFile().exists()) {
+      throw new RuntimeException("Could not find scanner directory: " + scannerPath);
+    }
+    this.scanInterval = scannerConfiguration.scanInterval();
+    this.registrationQueue = Objects.requireNonNull(registrationQueue,
+        "eventQueue must not be null");
+    this.usersErrorDirectory = globalConfig.usersErrorDirectory();
+  }
+
+  @Override
+  public void run() {
+    log.info("Started scanning '%s'".formatted(scannerPath));
+    while (!Thread.interrupted()) {
+      try {
+        var userFolderIterator = Arrays.stream(
+                Objects.requireNonNull(scannerPath.toFile().listFiles())).filter(File::isDirectory)
+            .toList().iterator();
+
+        while (userFolderIterator.hasNext()) {
+          fetchRegistrationDirectory(userFolderIterator.next().toPath()).ifPresent(
+              this::addRegistrationDirectory);
+        }
+
+        List<RegistrationRequest> requests = detectDataForRegistration();
+        for (RegistrationRequest request : requests) {
+          if (submittedRequests.contains(request)) {
+            log.info("Skipping registration request '{}'", request);
+            continue;
+          }
+          registrationQueue.add(request);
+          submittedRequests.add(request);
+          log.info("New registration requested: %s".formatted(request));
+        }
+        removePathZombies();
+        Thread.sleep(scanInterval);
+      } catch (InterruptedException e) {
+        interrupt();
+      }
+    }
+    log.info("Stopped scanning '%s'".formatted(scannerPath));
+  }
+
+  private List<RegistrationRequest> detectDataForRegistration() {
+    return userProcessDirectories.parallelStream()
+        .map(Path::toFile)
+        .map(file -> createRequests(file.listFiles(), file.toPath())).flatMap(
+            Collection::stream).toList();
+  }
+
+  private boolean isNotErrorFolder(File folder) {
+    return !folder.getName().equals(usersErrorDirectory.getFileName());
```

Comment:
doesn't this check the error folder itself and not its content? this seems like a very specific fail case (user moves the error folder?)